### PR TITLE
activate-system: Activate user environment too

### LINKED ISSUE #420: [help] launchd.envVariables after reboot
Setting `launchd.envVariables` does not seem to survive after reboot, the script is generated to `activate`, which will not be executed in launch daemon `org.nixos.activate-system`?
Is there any way to set launchd environment variables other than manually writing another launch agent?
```nix
  launchd.envVariables = { COLORTERM = "truecolor"; };
```

### PR DESCRIPTION
**Pull request by booxter**
Closes #420

### CODE REVIEW

File: modules/services/activate-system/default.nix [hunk 1]
```diff
@@ -42,5 +42,10 @@ in
       serviceConfig.KeepAlive.SuccessfulExit = false;
     };
 
+    launchd.user.agents.activate-user = {
+      command = "/run/current-system/activate-user";
+      serviceConfig.RunAtLoad = true;
+      serviceConfig.KeepAlive.SuccessfulExit = false;
+    };
```
Comment (Enzime) [inline modules/services/activate-system/default.nix:49]: This will run `activate-user` as every user on a machine which is not ideal

**Review by Enzime — CHANGES_REQUESTED** [review]
Thanks for the PR, but currently nix-darwin doesn't have the concept of primary user or per-user activation which is necessary before something like this should be added

see: https://github.com/LnL7/nix-darwin/pull/1016#issuecomment-2285910068

File: modules/services/activate-system/default.nix (hunk 1 above)
Comment (booxter) [reply modules/services/activate-system/default.nix:49]: Thanks for this comment, I didn't realize it.

On a related note, I am now realizing that there's also `launchd.agents` module in `home-manager`. Is it a different module? Does it mean I can achieve the same effect (scoped to my user) by defining it there instead? If so, then I think I may not need this patch in `nix-darwin` at all. I can instead define the service that `launchctl setenv` in HM instead. (That's the only part of the `activate-user` script I actually care about here.)

Comment (Enzime) [reply modules/services/activate-system/default.nix:49]: Yeah, putting it in home-manager will definitely work better as it’ll be specific to your user

**Comment by booxter** [comment]
Abandoning as per comments, will implement this in HM instead then.

**Pull request closed without merging**